mplc/multi_partner_learning.py

```diff
@@ -102,14 +101,19 @@ def compute_test_score_for_single_partner(self, partner):
                        self.epoch_count,
                        )
 
+        model_evaluation_val_data = self.evaluate_model(model, self.val_data)
+        current_val_loss = model_evaluation_val_data[0]
+        self.loss_collective_models.append(current_val_loss)
+
         # Evaluate trained model
         model_evaluation_test_data = self.evaluate_model(model, self.test_data)
         logger.info(f"   Model evaluation on test data: "
                     f"{list(zip(model.metrics_names, ['%.3f' % elem for elem in model_evaluation_test_data]))}")
 
         # Save model score on test data
         self.test_score = model_evaluation_test_data[1]  # 0 is for the loss
-        self.nb_epochs_done = (es.stopped_epoch + 1) if es.stopped_epoch != 0 else self.epoch_count
+        self.nb_epochs_done = (es.stopped_epoch + 1) if (self.is_early_stopping
+                                                         and es.stopped_epoch != 0) else self.epoch_count
```

Comment:
cf my previous comment:
https://github.com/SubstraFoundation/distributed-learning-contributivity/pull/241#discussion_r499772512

Reply:
I think I missed something
In the case of there is an early stopping, and it was triggered (ie es.stopped_epoch > 0), the number of epoch done is equal to es.stopped_epoch right ?
At the opposite, if there is no es, or if it was not triggered, the number of epoch done should be epoch_count ?